guvnor-m2repo-editor/guvnor-m2repo-editor-backend/src/main/java/org/guvnor/m2repo/backend/server/helpers/HttpGetHelper.java

```diff
@@ -46,35 +46,37 @@
     @Inject
     private GuvnorM2Repository repository;
 
-    public void handle( final HttpServletRequest request,
-                        final HttpServletResponse response,
-                        final ServletContext context ) throws IOException {
+    public void handle(final HttpServletRequest request,
+                       final HttpServletResponse response,
+                       final ServletContext context) throws IOException {
         String requestedFile = request.getPathInfo();
 
-        if ( requestedFile == null ) {
-            response.sendError( HttpServletResponse.SC_NOT_FOUND );
+        if (requestedFile == null) {
+            response.sendError(HttpServletResponse.SC_NOT_FOUND);
             return;
         }
 
-        requestedFile = URLDecoder.decode( requestedFile,
-                                           "UTF-8" );
+        requestedFile = URLDecoder.decode(requestedFile,
+                                          "UTF-8");
+
+        String repositoryName = request.getParameter("repository");
```

Comment:
Can you guarantee this?

When "browsing" the Workbench's repository in (for example, your web browser) and they enter a URL like ```http://server/kie-wb/maven2/org/cheese/porcelli/caps.jar``` an HTTP ```GET``` will be invoked which does not include a repository name. In addition if a User has added the workbench's Artifact Repository as a remote repository to another project's pom Maven will issue HTTP GETs when resolving dependencies from this remote. Those requests will not contain a repository name either.